file_reader_block.py

```diff
@@ -10,6 +12,26 @@ class FileReader(Block):
     Input signal trigger a read from a file. The contents of that file are
     placed on the notified signal in a configurable attribute.
     """
+    file = ExpressionProperty(title='File', default='/tmp/file.txt')
+    file_attr = StringProperty(title='File Field', default='file')
+    contents_attr = StringProperty(title='File Contents Field',
+                                   default='contents')
+    version = VersionProperty('0.1.0')
 
     def process_signals(self, signals, input_id='default'):
-        pass
+        for signal in signals:
+            try:
+                file = self.file(signal)
+            except:
+                self._logger.exception('Failed to evaluate file location')
+                continue
+            try:
+                self._logger.debug('Opening file: {}'.format(file))
+                with open(file) as openfile:
```

Comment:
We have a function `NIOEnvironment.get_path(filename)` that will allow you to specify either a path relative to your project or an absolute path. That might be better here, I can see wanting to read from a project file